File: python/Dosen.py
from SivitasAkademik import SivitasAkademik
from ProgramStudi import ProgramStudi

class Dosen (SivitasAkademik):
  __nip = ''# atribut NIP
  __prodi = ProgramStudi()# atribut prodi
  __fakultas = ''# atribut fakultas
  __pendTerakhir = ''# atribut fakultas
  __keahlian = ''# atribut fakultas


  def __init__(self, nik = '', nama = '', gender = '', nip = '',fakultas = '', prodi = ProgramStudi(),  pendTerakhir = '', keahlian = '', asalUniv = '', emailEdu = ''):
    # inisiasi dengan value awal
    super().__init__(nik, nama, gender, asalUniv, emailEdu)
    self.__nip = nip
    self.__prodi = prodi
    self.__fakultas = fakultas
    self.__pendTerakhir = pendTerakhir
    self.__keahlian = keahlian
  

  def getNip(self):
    # return NIP mahasiswa
    return self.__nip
  
  def getProdi(self):
    # return prodi
    return self.__prodi
  
  def getFakultas(self):
    #return fakultas
    return self.__fakultas
  
  def getPendTerakhir(self):
    #return pendTerakhir
    return self.__pendTerakhir
  
  def getKeahlian(self):
    #return keahlian
    return self.__keahlian
  
  def setNip(self, nip):
    # menggati value NIP
    self.__nip = nip
  
  def setProdi(self,prodi):
    # mengganti value prodi
    self.__prodi = prodi
  
  def setFakultas(self, fakultas):
    # mengganti value fakultas
    self.__fakultas = fakultas
    
  def setPendTerakhir(self, pendTerakhir):
    # mengganti value penTerakhir
    self.__pendTerakhir = pendTerakhir
    
  def setKeahlian(self, keahlian):
    # mengganti value keahlian
    self.__keahlian = keahlian

  

File: c++/Dosen.cpp
class Dosen : public SivitasAkademik {
  private:
    string nip; // atribut NIP
    ProgramStudi prodi; // atribut prodi
    string fakultas; // atribut fakultas
    string pendTerakhir; // atribut pendTerakhir
    string keahlian; // atribut keahlian

  public:
    Dosen(){
      this->nip = "";
      this->prodi = ProgramStudi();
      this->fakultas = "";
      this->pendTerakhir = "";
      this->keahlian = "";
    }
    Dosen(string nik, string nama, string gender, string nip, string fakultas, ProgramStudi prodi, string pendTerakhir, string keahlian, string asalUniv, string emailEdu): SivitasAkademik(nik, nama, gender, asalUniv, emailEdu){
      this->nip = nip;
      this->prodi = prodi;
      this->fakultas = fakultas;
      this->pendTerakhir = pendTerakhir;
      this->keahlian = keahlian;
    }

    string getNip() {
        // return NIP dosen
        return this->nip;
    }

    ProgramStudi getProdi() {
        // return prodi
        return this->prodi;
    }

    string getFakultas() {
        // return fakultas
        return this->fakultas;
    }

    string getPendTerakhir() {
        // return pendTerakhir
        return this->pendTerakhir;
    }

    string getKeahlian() {
        // return keahlian
        return this->keahlian;
    }

    void setNip(string nip) {
        // mengganti value NIP
        this->nip = nip;
    }

    void setProdi(ProgramStudi prodi) {
        // mengganti value prodi
        this->prodi = prodi;
    }

    void setFakultas(string fakultas) {
        // mengganti value fakultas
        this->fakultas = fakultas;
    }

    void setPendTerakhir(string pendTerakhir) {
        // mengganti value pendTerakhir
        this->pendTerakhir = pendTerakhir;
    }

    void setKeahlian(string keahlian) {
        // mengganti value keahlian
        this->keahlian = keahlian;
    }
};
